fitsio/fitslib.py

```diff
@@ -2505,17 +2532,22 @@ def _read_rec_with_var(self, colnums, rows, dtype, offsets, isvar, vstorage):
                     # storing in object array
                     # get references to each, no copy made
                     for irow,item in enumerate(dlist):
+                        if IS_PY3 and isinstance(item, bytes):
+                            item = item.decode('ascii')
```

Comment:
I'm not so sure about testing each object in an array. This might muck with other things or cause random decoding errors on things not meant to be decoded. In general, we cannot know if a bytes object is supposed to be a string or something else.

Reply:
after discussion with @esheldon I think we are good on this. The only thing in the C lib that comes back out as bytes are TSTRING items and those should always be strings.